modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/file_info_query.rb

```diff
@@ -34,58 +34,80 @@ module StorageInteraction
             end
 
-            Auth[auth_strategy].call(storage: @storage) do |http|
-              @delegate.call(http:, drive_item_id: file_id, fields: FIELDS).map(&storage_file_infos)
+            user_bound_result = Authentication[auth_strategy].call(storage: @storage) do |http|
+              @drive_item_query.call(http:, drive_item_id: file_id, fields: FIELDS)
+            end
+
+            user_bound_result.on_success { |sr| return ServiceResult.success(result: storage_file_infos(sr.result)) }
+            user_bound_result.on_failure do |sr|
+              return sr unless sr.result == :not_found
+
+              return admin_query(file_id)
             end
           end
 
           private
 
-          # rubocop:disable Metrics/AbcSize
-          def storage_file_infos
-            ->(json) do
-              StorageFileInfo.new(
-                status: "ok",
-                status_code: 200,
-                id: json[:id],
-                name: json[:name],
-                last_modified_at: Time.zone.parse(json.dig(:fileSystemInfo, :lastModifiedDateTime)),
-                created_at: Time.zone.parse(json.dig(:fileSystemInfo, :createdDateTime)),
-                mime_type: Util.mime_type(json),
-                size: json[:size],
-                owner_name: json.dig(:createdBy, :user, :displayName),
-                owner_id: json.dig(:createdBy, :user, :id),
-                trashed: false,
-                last_modified_by_name: json.dig(:lastModifiedBy, :user, :displayName),
-                last_modified_by_id: json.dig(:lastModifiedBy, :user, :id),
-                permissions: nil,
-                location: Util.extract_location(json[:parentReference], json[:name])
+          def admin_query(file_id)
+            admin_result = Authentication[userless_strategy].call(storage: @storage) do |http|
+              @drive_item_query.call(http:, drive_item_id: file_id, fields: FIELDS)
+            end
+
+            admin_result.on_success do |admin_query|
+              return ServiceResult.success(
+                result: storage_file_infos(admin_query.result, status: "forbidden", status_code: 403)
               )
             end
           end
 
-          # rubocop:enable Metrics/AbcSize
+          def userless_strategy = Registry.resolve("one_drive.authentication.userless").call
+
+          def storage_file_infos(json, status: "ok", status_code: 200)
+            info = StorageFileInfo.new(
+              status:,
+              status_code:,
+              id: json[:id],
+              name: json[:name],
+              mime_type: Util.mime_type(json),
+              size: json[:size],
+              owner_name: json.dig(:createdBy, :user, :displayName),
+              owner_id: json.dig(:createdBy, :user, :id),
+              trashed: false,
+              permissions: nil,
+              location: Util.extract_location(json[:parentReference], json[:name])
+            )
+
+            add_file_change_info(info, json)
```

Comment:
🟡 is this deconstruction really necessary, just to please the linter?? I was at least 30 seconds looking at it and asked myself "Why?". If the linter rule is violated here, but there is a good reason to violate it (as in constructing this object), we should disable it locally, but not forcefully deconstruct our methods.

Reply:
It started as I was trying to figure if it was worth the effort of building a better way to transform the JSON, but this was a leftover. =/